coda/coda_validate/tests/test_utils.py

```diff
@@ -0,0 +1,154 @@
+import pytest
+from lxml import etree, objectify
+
+from .. import views, factories
+from ..models import Validate
+
+
+VALIDATE_XML = '{http://digital2.library.unt.edu/coda/validatexml/}validate'
+
+
+@pytest.mark.parametrize('month, expected', [
+    (1, 31),
+    (2, 28),
+    (3, 31),
+    (4, 30),
+    (5, 31),
+    (6, 30),
+    (7, 31),
+    (8, 31),
+    (9, 30),
+    (10, 31),
+    (11, 30),
+    (12, 31)
+])
+def test_last_day_of_month(month, expected):
+    assert views.last_day_of_month(2015, month) == expected
+
+
+@pytest.fixture
+def validate_feed():
+    return """<?xml version="1.0"?>
+        <entry xmlns="http://www.w3.org/2005/Atom">
+        <title>ark:/00001/codajom1</title>
+        <id>http://example.com/APP/validate/ark:/00001/codajom1/</id>
+        <updated>2015-08-17T17:13:07Z</updated>
+        <author>
+            <name>Coda</name>
+            <uri>http://digital2.library.unt.edu/name/nm0004311/</uri>
+        </author>
+        <content type="application/xml">
+            <v:validate xmlns:v="http://digital2.library.unt.edu/coda/validatexml/">
+                <v:identifier>ark:/00001/codajom1</v:identifier>
+                <v:last_verified>2015-01-01 12:11:43</v:last_verified>
+                <v:last_verified_status>Passed</v:last_verified_status>
+                <v:priority_change_date>2000-01-01 00:00:00</v:priority_change_date>
+                <v:priority>1</v:priority>
+                <v:server>arch01.example.com</v:server>
+            </v:validate>
+        </content>
+        </entry>
+    """
+
+
+def test_xmlToValidateObject(validate_feed):
+    validate = views.xmlToValidateObject(validate_feed)
+    xml_obj = objectify.fromstring(validate_feed)
+    validate_xml = xml_obj.content[VALIDATE_XML]
+
+    assert validate_xml.identifier == validate.identifier
+    assert validate_xml.last_verified == str(validate.last_verified)
+    assert validate_xml.last_verified_status == str(validate.last_verified_status)
+    assert validate_xml.priority_change_date == str(validate.priority_change_date)
+    assert validate_xml.priority.text == validate.priority
+    assert validate_xml.server == validate.server
+
+
+@pytest.mark.xfail(reason='Exception will never be raised directly from the function.')
+def test_xmlToValidateObject_raises_exception():
+    assert 0
```

Comment:
this is more to document that something needs to change about this function rather than to create a test that will pass in the future.

(we discussed this in person. just putting it here for @ldko)